pkgs/development/compilers/dotnet/patch-nupkgs.nix

```diff
@@ -31,45 +31,54 @@ writeShellScriptBin "patch-nupkgs" (
   ''
     set -euo pipefail
     shopt -s nullglob
-    isELF() {
-        local fn="$1"
-        local fd
-        local magic
-        exec {fd}< "$fn"
-        read -r -n 4 -u "$fd" magic
-        exec {fd}<&-
-        if [ "$magic" = $'\177ELF' ]; then return 0; else return 1; fi
-    }
     cd "$1"
   ''
   + lib.optionalString stdenv.hostPlatform.isLinux ''
+    isNonStaticExecutableELF() {
+      local magic
+      local e_type
+      local interpreter
+
+      if magic=$(od --address-radix=n --read-bytes=4 --format=c "$1") \
+        && [ "$magic" = " 177   E   L   F" ] \
+        && e_type=$(od --address-radix=n --skip-bytes=0x10 --read-bytes=2 --format=x2 "$1") \
+        && interpreter=$(${patchelf}/bin/patchelf --print-interpreter "$1") \
+        && [ "$e_type" != " 0002" -o -n "$interpreter" ]; then return 0; else return 1; fi
+    }
+
     for x in */* */*; do
       # .nupkg.metadata is written last, so we know the packages is complete
       [[ -d "$x" ]] && [[ -f "$x"/.nupkg.metadata ]] \
         && [[ ! -f "$x"/.nix-patched ]] || continue
       echo "Patching package $x"
       find "$x" -type f -print0 | while IFS= read -rd "" p; do
         if [[ "$p" != *.nix-patched ]] \
-          && isELF "$p" \
-          && ${patchelf}/bin/patchelf --print-interpreter "$p" &>/dev/null; then
+          && isNonStaticExecutableELF "$p" \
```

Comment:
Do we still need `isNonStaticExecutableELF` after the other changes?  `isELF` was borrowed from elsewhere in nixpkgs, and I was hoping to move it to a common location at some point.

Reply:
No, we could revert back to the original `isELF`.